hathor/consensus.py

```diff
@@ -242,14 +243,46 @@ def update_voided_info(self, block: Block) -> None:
                     meta = block.get_metadata()
                     if not meta.voided_by:
                         storage._best_block_tips = [block.hash]
+                        self.log.debug('index new winner block', height=meta.height, block=block.hash_hex)
+                        # We update the height cache index with the new winner chain
+                        self.update_block_height_cache_new_chain(meta.height, block)
                 else:
                     storage._best_block_tips = [blk.hash for blk in heads]
+                    # XXX Is it safe to select one of the heads?
```

Comment:
I'm not sure this is the best approach. If there are more than one head, all of them will be voided. So maybe the best here is to get the first parent block that is executed.

Reply:
I'm not sure either, but I think we need to make a choice here because the block height index will expect the parent block to exist when we get there.

Reply:
The parent block must always exist but it might be voided.

Reply:
I feel we can open an issue to handle it later. What do you think?

Reply:
Moved to #238

Reply:
Moved to #238